### ISSUE TITLE
using --inspect crashes on restart

### ISSUE THREAD

**Issue by andrewgbliss** [JacksonGariety/gulp-nodemon#139]
Here is my nodemon gulpfile. 

`
gulp.task('start:dev', done => {
  let started = false;
  nodemon({
    script: path.join(__dirname, 'index.js'),
    ext: 'js json',
    nodeArgs: ['--inspect=0.0.0.0:9229'],
    watch: path.join(__dirname, 'express'),
    legacyWatch: true,
    delay: 2000
  })
  .on('start', () => {
    // to avoid nodemon being started multiple times
    if (!started) {
      setTimeout(() => done(), 100);
      started = true;
    }
  });
});
`

Every time I save a file nodemon restarts as it normally would. But it also crashes with this error message:

`
Starting inspector on 0.0.0.0:9229 failed: address already in use
app crashed - waiting for file changes before starting...
`

If I then save the file again it restarts it just fine. Is there a way to wait for the 9229 port to release before it starts again? It's really annoying to save a file twice just to get it to work.

**Comment by rluncasu** [comment]
Same here, I'm starting it with 
```sh
nodemon --inspect=0.0.0.0:9229 /app/server.js
```
inside a container based on
```
FROM node:8.1.0
```
with a shared volume on sources. Whenever I save a file on the sources volume, nodemon restarts but crashes with 
```
[nodemon] restarting due to changes...
  nodemon bus emit: restart +13s
  nodemon bus.emit(exit) via SIGUSR2 +14ms
  nodemon bus emit: exit +0ms
[nodemon] starting `node --inspect=0.0.0.0:9229 /app/server.js`
  nodemon spawning node --inspect=0.0.0.0:9229 /app/server.js +1ms
  nodemon bus emit: start +7ms
  nodemon bus new listener: exit (1) +1ms
  nodemon start watch on: *.* +0ms
  nodemon:watch early exit on watch, still watching (1) +1ms
Starting inspector on 0.0.0.0:9229 failed: address already in use
[nodemon] app crashed - waiting for file changes before starting...
```

Re-saving the same file, again restarts but works ok this time.

**Comment by megakoresh** [comment]
Yeah same problem, using sails here + nodemon. Doing the following:
`nodemon --inspect app.js`
produces
`error: Gulp :: Starting inspector on 127.0.0.1:9229 failed: address already in use`
As the error shows up before any tasks are even attempted, I think this must be a deeper issue.

**Comment by yujiahaol68** [comment]
Just kill inspector and start inspector again
here is our team's solution in our `package.json`
The same in gulp task. You have to kill it first..

```
"inspect": "kill-port --port 9229 && node --inspect=0.0.0.0:9229 build/startup.js",
"start_watch_inspect": "nodemon --delay 80ms --watch build/ build/startup.js --exec 'npm run inspect'",
```

**Comment by poonam-dev** [comment]
![error](https://user-images.githubusercontent.com/31477765/29875542-fba09e32-8db7-11e7-9bc5-42701b61e86d.png)

please solve this

**Comment by yujiahaol68** [comment]
Please try not to use nodemon to enable debug mode. Try to use `node --inspect=...`

**Comment by poonam-dev** [comment]
![error1](https://user-images.githubusercontent.com/31477765/29877211-9d580b76-8dbc-11e7-9552-7b841e562774.png)

now generate this error

**Comment by poonam-dev** [comment]
for put  node --inspect=... at the place of nodemon --debug

**Comment by peeyushsrivastava** [comment]
Actually it's getting started in background so you need to search for process which you have tried to run with command like nodemon --inspect app.js , so check if node server is running or not. 

ps ax | grep node

then killall -9 node and then try this command and it will work.

**Comment by wzup** [comment]
@yujiahaol68 
> kill-port: command not found

**Comment by Robinfr** [comment]
No solution for this? It's really annoying as you don't even get output anymore when it fails..

**Comment by SenP** [comment]
This seems to be an irritating bug with no clear solution yet
